### PR TITLE
fix(ingest): remove extraneous platform configs

### PR DESCRIPTION
**Pull request by hsheth2**
Also refactors to use mixin classes and flatten config inheritance hierarchy.

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/configuration/source_common.py [hunk 1]
```diff
@@ -53,13 +49,13 @@ def env_must_be_one_of(cls, v: str) -> str:
         return v.upper()
 
 
-class DatasetSourceConfigBase(PlatformSourceConfigBase, EnvBasedSourceConfigBase):
+class DatasetSourceConfigMixin(PlatformInstanceConfigMixin, EnvConfigMixin):
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/configuration/source_common.py:52]: Are these all just renames to use the MixIn nomenclature?

File: metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_config.py [hunk 2]
```diff
@@ -266,13 +267,6 @@ def get_sql_alchemy_url(self, run_on_compute: bool = False) -> str:
...
-    def platform_is_always_bigquery(cls, v):
-        return "bigquery"
-
-    @validator("platform_instance")
-    def bigquery_doesnt_need_platform_instance(cls, v):
-        if v is not None:
-            raise ConfigurationError(
-                "BigQuery project ids are globally unique. You do not need to specify a platform instance."
-            )
+    platform_instance_not_supported_for_bigquery = pydantic_removed_field(
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/bigquery_v2/bigquery_config.py:270]: Nice good to know we don't even need this!

File: metadata-ingestion/src/datahub/ingestion/source/sql/redshift.py [hunk 3]
```diff
@@ -170,10 +169,6 @@ class RedshiftConfig(
         description="Which table lineage collector mode to use. Available modes are: [stl_scan_based, sql_based, mixed]",
     )
 
-    @pydantic.validator("platform")
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/sql/redshift.py:173]: what is this? we are just removing platform from bigquery completely?

**Review by jjoyce0510 — APPROVED** [review]
A few questions, but overall LGTM

File: metadata-ingestion/src/datahub/configuration/source_common.py (hunk 1 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/configuration/source_common.py:52]: renames + removing the mixins in some cases

File: metadata-ingestion/src/datahub/ingestion/source/sql/redshift.py (hunk 3 above)
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/sql/redshift.py:173]: Yup - if the `platform` needs to always be "bigquery", why is it even a config? Also it turned out that we weren't really ever using that config anyways

**Pull request merged**